### PR TITLE
Add previews for Reviews blocks

### LINKED ISSUE #1029: Reviews by Product - add preview
Add an example preview to the block in the Inspector.

### PR DESCRIPTION
**Pull request by Aljullu**
Fixes #1029.

### Screenshots
Peek 2019-10-21 16-12

### How to test the changes in this Pull Request:

1. Create a new post and, using the top bar inserter, hover the blocks _All Reviews_, _Reviews by Product_ and _Reviews by Category_.
2. Verify they show a preview.

### Changelog

> Add previews to All Reviews, Reviews by Product and Reviews by Category blocks.

### CODE REVIEW

File: assets/js/base/hocs/with-reviews.js [hunk 1]
```diff
@@ -13,14 +13,18 @@ import { formatError } from '../utils/errors.js';
...
 
 			this.state = {
 				error: null,
 				loading: true,
-				reviews: [],
-				totalReviews: 0,
+				reviews: this.isPreview ? props.attributes.previewReviews : [],
+				totalReviews: this.isPreview
+					? props.attributes.previewReviews.length
+					: 0,
```
Comment (nerrad) [inline assets/js/base/hocs/with-reviews.js]: would be a good opportunity here to refactor this to class properties and get rid of the constructor?

File: assets/js/base/hocs/with-reviews.js [hunk 2]
```diff
@@ -13,14 +13,18 @@ import { formatError } from '../utils/errors.js';
 
 const withReviews = ( OriginalComponent ) => {
 	class WrappedComponent extends Component {
-		constructor() {
-			super( ...arguments );
+		constructor( props ) {
+			super( props );
+
+			this.isPreview = Boolean( props.attributes.previewReviews );
```
Comment (nerrad) [inline assets/js/base/hocs/with-reviews.js]: What about just doing `!! props.attributes.previewReviews` here?

File: assets/js/base/hocs/with-reviews.js (hunk 1 above)
Comment (Aljullu) [reply assets/js/base/hocs/with-reviews.js]: Good idea! Done in 9ec372240baa707a384c6ac8bc400bfa8342d934.

File: assets/js/base/hocs/with-reviews.js (hunk 2 above)
Comment (Aljullu) [reply assets/js/base/hocs/with-reviews.js]: I usually prefer doing things easier for newcomers and I think `Boolean( ... )` is easier to understand than `!! ...`. But double negation seems to be a bit more performant, so I'm ok either way. Changed in 4e8dee42d0b980b22ba747ecff620f4bda387171.

**Comment by Aljullu** [comment]
Thanks for the review @nerrad! Can you take another look?

**Pull request merged**